detection_rules/rule_formatter.py

```diff
@@ -74,6 +74,31 @@ class NonformattedField(str):
     """Non-formatting class."""
 
 
+def preserve_formatting_for_fields(data: OrderedDict, fields_to_preserve: list) -> OrderedDict:
+    """Preserve formatting for specified nested fields in an action."""
+
+    def apply_preservation(target: OrderedDict, keys: list):
+        """Apply NonformattedField preservation based on keys path."""
+        for key in keys[:-1]:
+            # Iterate to the  key, diving into nested dictionaries
+            if key in target and isinstance(target[key], dict):
+                target = target[key]
+            else:
+                # Cannot preserve formatting for missing or non-dict intermediate
+                return
```

Comment:
Should we warn here that the preservation failed?

Reply:
This should never happen TBH. It's not a dynamic field.